src/instructlab/data/list.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+from typing import List, Tuple, TypedDict
+import logging
+import time
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetListing(TypedDict):
+    """
+    Represents a single dataset listing.
+    """
+
+    filename: str
+    created_at: str
+    file_size: str
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: int) -> Tuple[float, str]:
+    magnitudes = ["KB", "MB", "GB"]
+    magnitude = "B"
+    adjusted_fsize = float(f_size)
+    for mag in magnitudes:
+        if adjusted_fsize >= 1024:
+            magnitude = mag
+            adjusted_fsize /= 1024
+        else:
+            return adjusted_fsize, magnitude
+    return adjusted_fsize, magnitude
```

Comment:
can we just make this a general utility function since it will be getting used more than once?

Reply:
Yes but I'm keeping it here for now to avoid a messy rebase because otherwise it's going to exist in the `model/list.py` file.

Reply:
I don't understand which messy rebase you are talking about...